### PR TITLE
tests/fuzzers/abi: add fuzzer for fuzzing package accounts/abi

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
This PR adds a fuzzer based on go-fuzz and gofuzz to thoroughly test the package `accounts/abi`
It uses randomness based on the input to create reproducible test cases. The corpus is currently very lacking so I will update it later. 

It found two bugs already:
- function `packElement` crashed on input that is not a byte (e.g. 	"\x9e")
- function`toGoType` crashes with indexOutOfBounds because of invalid casting of the offset

### CODE REVIEW

File: tests/fuzzers/abi/abifuzzer.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+// +build gofuzz !gofuzz
```
Comment (MariusVanDerWijden) [inline tests/fuzzers/abi/abifuzzer.go:17]: Do something about this tautology

**Comment by Precha23** [comment]
Ok

**Comment by holiman** [comment]
This one has been left unreiewed too long! I think it looks good - just a rebase needed right?

**Comment by MariusVanDerWijden** [comment]
> This one has been left unreiewed too long! I think it looks good - just a rebase needed right?

@holiman Well yeah, but this pr also contains a crasher that we haven't fixed yet cc @gballet

**Comment by MariusVanDerWijden** [comment]
So the crasher was basically that `abi.Unpack` is a little bit more powerful than `abi.Pack`. Unpack can successfully unpack an uint8 into an int8. Pack however checks first if they are of the same type and returns an error. I spent several hours today trying to make Pack accept a uint8 to int8 packing but it breaks other code.

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**